blog/2024-08-07-introducing-secrets-how-wasmcloud-solves-for-secrets-in-distributed-environments/index.mdx

```diff
@@ -0,0 +1,203 @@
+---
+title: "wasmCloud 1.1: Secure, pluggable WebAssembly secrets with Vault, K8s secrets, and more"
+authors: [brooksmtownsend]
+slug: secure-pluggable-webassembly-secrets-with-vault-k8s-secrets
+image: './images/secure-pluggable-wasm-secrets.png'
+date: "2024-08-07"
+description: ""
+---
+
+![Introducing secrets](./images/secure-pluggable-wasm-secrets.png)
+
+In wasmCloud 1.1, we're excited to introduce what is far and away our most requested feature: **secrets**. Now applications on wasmCloud can request secrets at runtime, leveraging a secrets API that can be used with the secret store of your choice. With a system that encrypts secrets in transit and never stores them on disk, we've doubled down on our Zero Trust security posture so teams can deploy to production with confidence. 
+
+## How secrets work in wasmCloud
+
+Secrets are a challenging problem in cloud native environments, so we spent a lot of time thinking about how to approach them in wasmCloud.  
+
+We wanted a better approach&mdash;one that leverages the NATS-powered lattice to deliver true encryption and avoid storing secrets on the [wasmCloud host](/docs/concepts/hosts) in any form that would be accessible to anything other than the component or provider requesting them. 
+
+We also wanted our secret system to be **pluggable**, so teams could use the secret store of their choice, whether that was HashiCorp Vault, cloud provider systems like AWS Secrets Manager, an in-house solution, or an open source store.
+
+To achieve all of this, we came up with an architecture that organizes secret requests and responses into a few key steps:
+
+* A component or capability provider **requests a secret** by reference to secret's location, without needing to present credential material. 
+* The request goes to a **secrets handler** which in turn requests the specified secret from the appropriate **secrets backend**.
+* Communication with a **secrets store** is mediated by a **secrets backend**.
+
+You can see how a component requests a secret in the diagram below:
+
+![diagram of secret flow when starting component with secret](../../static/img/secrets-diagram.png)
+
+If we zoom into an individual component level, the standard flow for a secret looks like this:
+
+![diagram](./images/component-secret.png)
+
+Let's walk through this process step-by-step:
+
+1. Component sends `wasmcloud:secrets/get` call to host
+2. Host returns secret reference
+3. Component sends `wasmcloud:secrets/reveal` call to host
+4. Host exposes secret from `secrecy` crate's in-memory store, returning value to component
+
+The secret itself is never logged or accessible in memory&mdash;it is encrypted with an [**xkey**](https://docs.nats.io/running-a-nats-service/configuration/securing_nats/auth_callout#encryption), a x25519 keypair, and held in-memory with Rust's [`secrecy` crate](https://crates.io/crates/secrecy). When you drop a secret, it is wiped in-memory. 
```

Comment:
```suggestion
The secret itself is never logged or accessible in memory&mdash; while in transit, the secret is encrypted with an [**xkey**](https://docs.nats.io/running-a-nats-service/configuration/securing_nats/auth_callout#encryption), a x25519 keypair, and when held in-memory by the Host, the host uses the Rust [`secrecy` crate](https://crates.io/crates/secrecy) to secure the contents. When you drop a secret, it is wiped in-memory. 
```

Reply:
This might be nuance that's not relevant for the blog post, but I figured I'd offer it in case it felt like it adds to the description.